### PR TITLE
[model] fix: remove npu flash attention sync

### PR DESCRIPTION
**Pull request by wey-code**
### What does this PR do?

> Place cu_seqlens, etc. on the CPU in advance to avoid triggering synchronization on the npu flash attention.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `core`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `
  - If this PR involves multiple modules, separate them with `,` like `[megatron, torch, liger-kernals,fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, torch, liger-kernals] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `make commit && make style && make quality`
- [x] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: veomni/models/auto.py [hunk 1]
```diff
@@ -124,4 +126,18 @@ def build_foundation_model(
         init_device=init_device,
     )
 
+    if is_torch_npu_available():
+        # for npu, set cu_seq_lens_q & cu_seq_lens_k to cpu
+        original_forward = model.forward
+
+        @functools.wraps(original_forward)
```
Comment (piyifan123) [inline veomni/models/auto.py:137]: nit: 

1, please add comment on why we need this due to https://github.com/pytorch/pytorch/blob/134179474539648ba7dee1317959529fbd0e7f89/torch/distributed/fsdp/_fully_shard/_fsdp_state.py#L130 

2, please add a log showing that we are overriding model forward for npu to ensure that FA kwargs are on cpu (to be used by npu_fused_attention which needs cpu fa kwargs

**Review by piyifan123 — APPROVED** [review]

File: veomni/models/auto.py (hunk 1 above)
Comment (wey-code) [reply veomni/models/auto.py:137]: Done~
